src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -3389,4 +3390,64 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
             let a: [u8; <error descr="Calls in constants are limited to constant functions, tuple structs and tuple variants [E0015]">foo</error>(1, <error descr="A non-constant value was used in a constant expression [E0435]">bar</error>)];
         }
     """)
+
+    fun `test E0517 placement repr C`() = checkErrors("""
+        #[repr(<error descr="C attribute should be applied to struct, enum, or union [E0517]">C</error>)]
```

Comment:
Hm, does it make sense to mark only `C` attribute since `repr` attribute can be applied for type item at all?

Reply:
There is no special meaning, just to detect whether there is an error when placed in `type`.

Reply:
PR has been merged, is there anything that needs to be fixed here?

Reply:
I see two things that we can improve here:
* if builtin `repr` attribute cannot be applied to a particular item (for example, for trait), there is no need to annotate arguments of the attr. Let's better annotate `repr` attribute itself in this case and provide quick fix to remove it. I know that compiler provides error for a particular argument but I don't know why.
* Potentially, you can create your own `repr` proc macro and provide custom rules for it. In this case, we shouldn't annotate anything because we don't know anything about these rules (at least for now). Note, the current implementation provides false positive for this case only when custom `repr` attribute is imported in current scope.
For example:
```rust
//- lib.rs
#[proc_macro_attribute]
pub fn repr(attr: TokenStream, item: TokenStream) -> TokenStream {
    item
}

//- main.rs
use lib::repr;

#[repr(C)] // <- we should provide neither E0517 nor E0552 in this case because compiler provides E0659
type Foo = i32;
```
Of course, it's rather rare case

Reply:
I want to address this situation, the simplest fix method.

```kotlin
if (attr.metaItem.name != "repr" || attr.metaItem.path?.reference?.resolve() != null) return
```

Reply:
Yep, looks perfect solution for the second case

Reply:
This issue also affects declarations such as `inline`, `feature`, etc.

Reply:
> if builtin repr attribute cannot be applied to a particular item (for example, for trait), there is no need to annotate arguments of the attr. Let's better annotate repr attribute itself in this case and provide quick fix to remove it. I know that compiler provides error for a particular argument but I don't know why.

Maybe for this situation(Multiple attribute value declarations)
```rust
#[repr(C, i32)]
struct Test(i32);
```
`C` can be placed on a `struct`, but `i32` is not

Reply:
> This issue also affects attribute declarations such as inline, feature, etc.

Yep. Ideally, we should check that there isn't custom attr proc macro in all such cases.
But as I mentioned above, it's a rather rare case

Reply:
> Maybe for this situation(Multiple attribute value declarations)

In your case, `repr` attribute may make sense for struct in general, but currently, it doesn't make sense for traits, type aliases, functions, etc. with any parameters so I think it's rather natural to suggest to drop the whole attribute instead of only its part.

In the mentioned case, the current implementation works as expected